internal/cmd/gocdk/add.go

```diff
@@ -16,10 +16,100 @@ package main
 
 import (
 	"context"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path"
+	"path/filepath"
+	"sort"
+	"strings"
 
 	"golang.org/x/xerrors"
 )
 
+type portableTypeInfo struct {
+	name       string // the portable type name
+	goDemoPath string // the path to the .go file to add to the project for static.Open
+	demoURL    string // the URL for the demo
+}
+
+var portableTypes = []*portableTypeInfo{
+	{
+		name:       "blob.Bucket",
+		goDemoPath: "/demo/blob.bucket/demo_blob_bucket.go",
+		demoURL:    "/demo/blob.bucket/",
+	},
+}
+
+// TODO(rvangent): Add tests for add(), including for each supported portableType.
+
 func add(ctx context.Context, pctx *processContext, args []string) error {
-	return xerrors.New("not implemented")
+	// Compute a sorted slice of available portable types for usage.
+	var avail []string
+	for _, pt := range portableTypes {
+		avail = append(avail, pt.name)
+	}
+	sort.Strings(avail)
+	usageMsg := fmt.Sprintf("gocdk add <%s>", strings.Join(avail, "|"))
+
+	f := newFlagSet(pctx, "add")
+	force := f.Bool("force", false, "re-add even the portable type even if it has already been added, overwriting previous files")
+	if err := f.Parse(args); xerrors.Is(err, flag.ErrHelp) {
+		return nil
+	} else if err != nil {
+		return usagef("%s: %w", usageMsg, err)
+	}
+
+	args = f.Args()
+	if len(args) != 1 {
+		return usagef("%s: expected 1 argument, got %d", usageMsg, len(args))
```

Comment:
Pointing out difference from original design doc: this command should take the form `gocdk add pt(ype?) TYPE`. I do wonder whether for simplicity it could be made into a single argument `add-ptype`, but I don't have thoughts on the ergonomics of this. @clausti for comment.

Reply:
What other things would `add` add?

Reply:
Biomes.

Reply:
I'm fine with changing `add ptype` and `add biome` to `add-ptype` and `add-biome` if that makes more sense to most of us.

Reply:
Done. I renamed the command (`add` to `add-ptype`) and the function (`add` to `addPortableType`), but didn't rename the file in this PR because it will make rebasing follow-on PRs hard. I added a TODOs to add `add-biome` and to rename the file.